File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/browsing/files/cached/disk/IDiskCacheDirectoryProvider.java
package com.lasthopesoftware.bluewater.client.browsing.files.cached.disk;

import com.lasthopesoftware.bluewater.client.browsing.files.cached.configuration.IDiskFileCacheConfiguration;

import java.io.File;

public interface IDiskCacheDirectoryProvider {
	File getDiskCacheDirectory(IDiskFileCacheConfiguration diskFileCacheConfiguration);
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/browsing/files/cached/disk/IDiskCacheDirectoryProvider.kt
package com.lasthopesoftware.bluewater.client.browsing.files.cached.disk

import com.lasthopesoftware.bluewater.client.browsing.library.repository.LibraryId
import java.io.File

interface IDiskCacheDirectoryProvider {
    fun getLibraryDiskCacheDirectory(libraryId: LibraryId): File?
    fun getRootDiskCacheDirectory(): File?
}
